### PR TITLE
[review-pending] docs(dev guide): component-styles - new dart prose and code; ts updates

### PR DESCRIPTION
**Pull request by chalin**
Starting point for Dart code was taken from #1171 (thanks @luisvt!).
Could not run e2e test on this TS example because of #1319.

### CODE REVIEW

File: public/docs/dart/latest/guide/component-styles.jade [hunk 1]
```diff
@@ -1 +1,40 @@
...
+extends ../../../ts/latest/guide/component-styles.jade
+
+block includes
+  include ../_util-fns
+
+//- TODO: consider adding material equivalent to TS Appendices 1 & 2 if relevant.
+
+block style-url-alert
+  :marked
+    The URL is ***relative to the Dart application package root*** which usually
```
Comment (zoechi) [inline public/docs/dart/latest/guide/component-styles.jade]: `styleUrls` paths worked relative to the component file until beta.17 for me in Dart

Comment (zoechi) [inline public/docs/dart/latest/guide/component-styles.jade]: Maybe I'm mixing something up. Here in the GitHub page it's not obvious what this is related to.

File: public/docs/ts/latest/guide/component-styles.jade [hunk 2]
```diff
@@ -156,37 +157,42 @@ a(id='loading-styles')
...
+      location of the `index.html` web page that hosts the application. 
+      The style file URL is *not* relative to the component file.
+      That's why the example URL begins `app/`.
+      See [Appendix 2](#relative-urls) to specify a URL relative to the 
+      component file.
+
+block module-bundlers
+  .l-sub-section
+    :marked
+      Users of module bundlers like Webpack may also use the `styles` attribute
```
Comment (zoechi) [inline public/docs/ts/latest/guide/component-styles.jade]: I guess this doesn't apply to Dart

File: public/docs/_examples/component-styles/dart/lib/hero.dart [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class Hero {
+  bool active = false;
+
+  String name;
```
Comment (thso) [inline public/docs/_examples/component-styles/dart/lib/hero.dart]: `final` and `final`

File: public/docs/_examples/component-styles/dart/lib/hero_app_component.dart [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+      <hero-app-main [hero]="hero"></hero-app-main>''',
+    styles: const ['h1 { font-weight: normal; }'],
+    directives: const [HeroAppMainComponent])
+class HeroAppComponent {
+// #enddocregion
+  Hero hero =
+      new Hero('Human Torch', ['Mister Fantastic', 'Invisible Woman', 'Thing']);
+
+  @HostBinding('class')
+  String get themeClass {
```
Comment (thso) [inline public/docs/_examples/component-styles/dart/lib/hero_app_component.dart]: Getter will be more elegant here.

File: public/docs/_examples/component-styles/dart/lib/hero_controls_component.dart [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+          background-color: white;
+          border: 1px solid #777;
+        }
+      </style>
+      <h3>Controls</h3>
+      <button (click)="activate()">Activate</button>''')
+class HeroControlsComponent {
+  @Input() Hero hero;
+
+  void activate() { hero.active = true; }
```
Comment (thso) [inline public/docs/_examples/component-styles/dart/lib/hero_controls_component.dart]: Do run `dartfmt` on newly created files.

File: public/docs/_examples/component-styles/dart/lib/hero_app_main_component.dart [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+import 'quest_summary_component.dart';
+
+@Component(
+    selector: 'hero-app-main',
+    template: '''
+      <quest-summary></quest-summary>
+      <hero-details [hero]="hero" [class.active]="hero.active">
+        <hero-controls [hero]="hero"></hero-controls>
+      </hero-details>
+    ''',
```
Comment (thso) [inline public/docs/_examples/component-styles/dart/lib/hero_app_main_component.dart]: Triple quotes after the last tag for consistency.

File: public/docs/_examples/component-styles/dart/lib/hero_details_component.dart [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+import 'hero_team_component.dart';
+
+// #docregion styleurls
+@Component(
+    selector: 'hero-details',
+    template: '''
+      <h2>{{hero.name}}</h2>
+      <hero-team [hero]=hero></hero-team>
+      <ng-content></ng-content>
+    ''',
```
Comment (thso) [inline public/docs/_examples/component-styles/dart/lib/hero_details_component.dart]: Ditto.

File: public/docs/_examples/component-styles/ts/app/hero-controls.component.ts [hunk 8]
```diff
@@ -17,10 +17,7 @@ import { Hero } from './hero';
 })
 // #enddocregion inlinestyles
 export class HeroControlsComponent {
-
   @Input() hero: Hero;
 
-  activate() {
-    this.hero.active = true;
-  }
+  activate() { this.hero.active = true; }
```
Comment (thso) [inline public/docs/_examples/component-styles/ts/app/hero-controls.component.ts]: Why this change? Please revert.

File: public/docs/ts/latest/guide/component-styles.jade (hunk 2 above)
Comment (thso) [inline public/docs/ts/latest/guide/component-styles.jade]: Yup, probably not.

File: public/docs/ts/latest/guide/component-styles.jade [hunk 9]
```diff
@@ -294,54 +304,61 @@ code-example(format='').
...
-  
-+makeExample('component-styles/ts/app/quest-summary.component.ts','', 'app/quest-summary.component.ts')
+    We can't do that by default. Angular can't find the files and throws an error:
+
+    `EXCEPTION: Failed to load quest-summary.component.html`
+
+    Why can't Angular calculate the HTML and CSS URLs from the component file's location? 
+
+    Unfortunately, that location is not readily known. 
+    Angular apps can be loaded in many ways: from individual files, from SystemJS packages, or
```
Comment (thso) [inline public/docs/ts/latest/guide/component-styles.jade]: These doesn't sound very Dart related.

File: public/docs/ts/latest/guide/component-styles.jade [hunk 10]
```diff
@@ -294,54 +304,61 @@ code-example(format='').
...
+    We can change the way Angular calculates the full URL be setting the component metadata's `moduleId` property.
+
+    If we knew the component file's base path, we'd set `moduleId` to that and 
+    let Angular construct the full URL from this base path plus the CSS and template file names.
+
+    Our challenge is to calculate the base path with minimal effort. 
+    If it's too hard, we shouldn't bother; we should just write the full path to the root and move on.
+
+    Fortunately, *certain* module loaders make it easy.
+    SystemJS (starting in v.0.19.19) sets a special `__moduleName` variable to the URL of the component file.
```
Comment (thso) [inline public/docs/ts/latest/guide/component-styles.jade]: I think we can kill this whole part of the text, in Dart you can specify paths relative to your component  folder. The Angular transformer will handle everything for you.

File: public/docs/ts/latest/guide/component-styles.jade (hunk 2 above)
Comment (chalin) [inline public/docs/ts/latest/guide/component-styles.jade]: Right. Whenever you see a line starting with
```jade
block foo
  some-stuff
```
Then look for the correspondingly named block in the Dart `.jade` file.

File: public/docs/_examples/component-styles/dart/lib/hero.dart (hunk 3 above)
Comment (chalin) [inline public/docs/_examples/component-styles/dart/lib/hero.dart]: done

File: public/docs/_examples/component-styles/dart/lib/hero_app_component.dart (hunk 4 above)
Comment (chalin) [inline public/docs/_examples/component-styles/dart/lib/hero_app_component.dart]: done

File: public/docs/_examples/component-styles/dart/lib/hero_controls_component.dart (hunk 5 above)
Comment (chalin) [inline public/docs/_examples/component-styles/dart/lib/hero_controls_component.dart]: done

File: public/docs/_examples/component-styles/dart/lib/hero_app_main_component.dart (hunk 6 above)
Comment (chalin) [inline public/docs/_examples/component-styles/dart/lib/hero_app_main_component.dart]: done

File: public/docs/_examples/component-styles/dart/lib/hero_details_component.dart (hunk 7 above)
Comment (chalin) [inline public/docs/_examples/component-styles/dart/lib/hero_details_component.dart]: done

File: public/docs/_examples/component-styles/ts/app/hero-controls.component.ts (hunk 8 above)
Comment (chalin) [inline public/docs/_examples/component-styles/ts/app/hero-controls.component.ts]: Because it matches compactness used in other TS examples.
Anyhow, change undone.

File: public/docs/ts/latest/guide/component-styles.jade (hunk 9 above)
Comment (chalin) [inline public/docs/ts/latest/guide/component-styles.jade]: It isn't This is the TS prose. It is difficult to see but this text is under the block `module-id` (see [Line 313](https://github.com/angular/angular.io/pull/1320/files#diff-452f86bc62408dcd7a93a8ce1ad3dacbR313)).
The corresponding Dart prose is [here](https://github.com/angular/angular.io/pull/1320/files#diff-1291d1db3d442dba259a6c4d2d3feea6R30).

File: public/docs/ts/latest/guide/component-styles.jade (hunk 10 above)
Comment (chalin) [inline public/docs/ts/latest/guide/component-styles.jade]: Ditto above. This is only part of the TS prose.

File: public/docs/dart/latest/guide/component-styles.jade (hunk 1 above)
Comment (chalin) [inline public/docs/dart/latest/guide/component-styles.jade]: You are correct, thanks. Fixed (though there seems to be a bug related to the styles actually being picked up properly; I'll try to isolated and file an issue soon).

**Comment by thso** [comment]
LGTM
